arangod/Aql/ClusterNodes.cpp

```diff
@@ -423,20 +423,26 @@ CostEstimate DistributeNode::estimateCost() const {
 
   while (node) {
     switch (node->getType()) {
-      case ENUMERATE_COLLECTION:
-        return castTo<EnumerateCollectionNode const*>(node)->collection();
       case INDEX:
-        return castTo<IndexNode const*>(node)->collection();
+      case ENUMERATE_COLLECTION: {
+        auto const* cNode = castTo<CollectionAccessingNode const*>(node);
+        if (!cNode->isUsedAsSatellite() && cNode->prototypeCollection() == nullptr) {
+          return cNode->collection();
+        }
+        break;
+      }
       case TRAVERSAL:
       case SHORTEST_PATH:
       case K_SHORTEST_PATHS:
         return castTo<GraphNode const*>(node)->collection();
```

Comment:
@maierlars 
I still think this is not correct for graphs. First, this must always be a local graph node (i.e. on a DBServer). Thus it is also a `CollectionAccessingNode`. You have to cast it and take the collection then, but only if it is not used as a satellite.